<?xml version='1.0'?>
<robot name="dd_robot">
    <!--
  <link name="dummy">
  </link>
-->
    <!-- Base Link -->
    <link name="base_link">
        <visual>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <box size="0.14 0.14 0.07"/>
            </geometry>
            <material name="blue">
                <color rgba="0 0.5 1 1"/>
            </material>
        </visual>
        <!-- Base collision, mass and inertia -->
        <collision>
            <origin xyz="0 0 0" rpy="0 0 0" />
            <geometry>
                <box size="0.14 0.14 0.07"/>
            </geometry>
        </collision>
        <inertial>
            <!-- COM experimentally determined -->
            <origin xyz="0 0 0" rpy="0 0 0" />
            <mass value="2.4"/>
            <!-- 2.4/2.6 kg for small/big battery pack -->
            <inertia ixx="0.019995" ixy="0.0" ixz="0.0"
                 iyy="0.019995" iyz="0.0" 
                 izz="0.03675" />
        </inertial>
        <!-- Caster -->
        <visual name="caster">
            <origin xyz="0.05 0 -0.041" rpy="0 0 0" />
            <geometry>
                <sphere radius="0.0175" />
            </geometry>
        </visual>
        <!-- Caster collision, mass and inertia -->
        <collision>
            <origin xyz="0.05 0 -0.041" rpy="0 0 0" />
            <geometry>
                <sphere radius="0.0175" />
            </geometry>
        </collision>
        <inertial>
            <mass value="0.5"/>
            <inertia ixx="0.0001" ixy="0.0" ixz="0.0" iyy="0.0001" iyz="0.0" izz="0.0001"/>
        </inertial>
    </link>
    <!--
  <joint name="dummy_joint" type="fixed">
     <parent link="dummy"/>
     <child link="base_link"/>
   </joint>
  -->
    <!-- Right Wheel -->
    <link name="right_wheel">
        <visual>
            <origin xyz="0 0 0" rpy="1.570795 0 0" />
            <geometry>
                <cylinder length="0.02" radius="0.07" />
            </geometry>
            <material name="black">
                <color rgba="0.05 0.05 0.05 1"/>
            </material>
        </visual>
        <!-- Right Wheel collision, mass and inertia -->
        <collision>
            <origin xyz="0 0 0" rpy="1.570795 0 0" />
            <geometry>
                <cylinder length="0.02" radius="0.07" />
            </geometry>
        </collision>
        <inertial>
            <origin xyz="0 0 0"/>
            <mass value="0.01" />
            <inertia ixx="0.001" ixy="0.0" ixz="0.0"
	        iyy="0.001" iyz="0.0" izz="0.001" />
        </inertial>
    </link>
    <!-- Right Wheel joint -->
    <joint name="joint1" type="continuous">
        <parent link="base_link"/>
        <child link="right_wheel"/>
        <origin xyz="0 -0.08 0" rpy="0 0 0" />
        <axis xyz="0 1 0" />
    </joint>
    <!-- Left Wheel -->
    <link name="left_wheel">
        <visual>
            <origin xyz="0 0 0" rpy="1.570795 0 0" />
            <geometry>
                <cylinder length="0.02" radius="0.07" />
            </geometry>
            <material name="black"/>
        </visual>
        <!-- Left Wheel collision, mass and inertia -->
        <collision>
            <origin xyz="0 0 0" rpy="1.570795 0 0" />
            <geometry>
                <cylinder length="0.02" radius="0.07" />
            </geometry>
        </collision>
        <inertial>
            <origin xyz="0 0 0"/>
            <mass value="0.01" />
            <inertia ixx="0.001" ixy="0.0" ixz="0.0"
	        iyy="0.001" iyz="0.0" izz="0.001" />
        </inertial>
    </link>
    <!-- Left Wheel joint -->
    <joint name="joint2" type="continuous">
        <parent link="base_link"/>
        <child link="left_wheel"/>
        <origin xyz="0 0.08 0" rpy="0 0 0" />
        <axis xyz="0 1 0" />
    </joint>
    <!-- imu -->
    <joint name="imu_joint" type="fixed">
        <axis xyz="0 0 0"/>
        <!-- 0 1 0 -->
        <origin xyz="0 0 0.036"/>
        <parent link="base_link"/>
        <child link="imu_link"/>
    </joint>
    <link name="imu_link">
        <inertial>
            <mass value="0.001"/>
            <origin rpy="0 0 1.57079632679" xyz="0.0221 -0.02 0.0016"/>
            <!-- <origin rpy="0 0 0" xyz="0 0 0"/> -->
            <inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.000001" iyz="0" izz="0.0001"/>
        </inertial>
        <visual>
            <origin rpy="0 0 1.57079632679" xyz="0.0221 -0.02 0.0016"/>
            <geometry>
                <box size="0.01 0.01 0.01"/>
            </geometry>
            <material name="xsens_orange">
                <color rgba="0.92 0.45 0.02 1.0"/>
            </material>
        </visual>
        <collision>
            <origin rpy="0 0 1.57079632679" xyz="0.0221 -0.02 0.0016"/>
            <geometry>
                <box size=".01 .01 .01"/>
            </geometry>
        </collision>
    </link>
</robot>